src/internet/dropbox/dropboxservice.cpp

```diff
@@ -267,6 +277,29 @@ QUrl DropboxService::GetStreamingUrlFromSongId(const QUrl& url) {
   QNetworkReply* reply = FetchContentUrl(url);
   WaitForSignal(reply, SIGNAL(finished()));
 
-  QJsonObject json_response = QJsonDocument::fromJson(reply->readAll()).object();
+  QJsonDocument document;
+  if (!ParseReply(reply, &document)) return QUrl();
+
+  QJsonObject json_response = document.object();
   return QUrl::fromEncoded(json_response["link"].toVariant().toByteArray());
 }
+
+// Note that this would look nicer if it returned a QJsonDocument, but move
```

Comment:
I don't think we care about the cost of a copy TBH.

Reply:
Changed the interface and also moved to the base class for future reuse.